tests/e2e/vmgroupmisc_test.go

```diff
@@ -224,3 +225,43 @@ func (s *vgBasicSuite) TestDSAccessPrivilegeForUserVG(c *C) {
 
 	misc.LogTestEnd(c.TestName())
 }
+
+// Test steps:
+// 1. Create a vm VM1 using govc
+// 2. Create a vmgroup and associate VM1 to it.
+// 2. Execute vmgroup ls command.
+// 3. Delete the vm that was added to the vmgroup.
+// 4. Again execute vmgroup ls command to verify command works fine.
+func (s *vgBasicSuite) TestDeleteVMFromVmgroup(c *C) {
+	misc.LogTestStart(c.TestName())
+	vmName := "VM_" + inputparams.GetRandomNumber()
+	vgName := "VG_" + inputparams.GetRandomNumber()
+	networkAdapterType := "vmxnet3"
+
+	// Create a vm - we need this to add to vmgroup and later on delete this vm
+	esx.CreateVM(vmName, s.config.Datastores[0], networkAdapterType)
+	c.Assert(esx.IsVMExist(vmName), Equals, true, Commentf("Failed to create VM - %s .", vmName, vgName))
+
+	out, err := admincli.CreateVMgroup(s.config.EsxHost, vgName, vmName, admincliconst.VMHomeDatastore)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Verify if vmgroup exists
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vgName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup %s does not exists.", vgName))
+
+	// Verify vm belongs to vmgroup
+	isVMPartofVg := admincli.IsVMInVmgroup(s.config.EsxHost, vmName, vgName)
+	c.Assert(isVMPartofVg, Equals, true, Commentf("VM %s does not belong to vmgroup %s .", vmName, vgName))
+
+	// Destroy the vm
+	esx.DestroyVM(vmName)
```

Comment:
govc does not returns error code. I wrote a util to check if vm exists. That I am using after creating a vm as well as after destroying vm.